src/zenml/zen_stores/schemas/step_run_schemas.py

```diff
@@ -212,11 +210,13 @@ def to_model(
         }
 
         if self.deployment is not None:
-            full_step_config = Step.parse_obj(
+            full_step_config = Step.model_validate(
```

Comment:
`Step.model_validate_json` maybe?

Reply:
I believe it is the correct usage here. If you remember, we added this as a form of backward compatibility. Step runs either have a `step_configuration` (a JSON serialized field), in which case we correctly call the `model_validate_json`, or `step_configurationS` where we have to do the JSON deserialization first, extract the relevant part of the config to the step, and then call `Step.model_validate()` with an already deserialized input.